OpenSim/Common/ComponentSocket.h

```diff
@@ -319,14 +319,14 @@ class OSIMCOMMON_API AbstractSocket {
 
     void setConnecteePathInternal(const std::string& name, int index) {
         using SimTK::isIndexInRange;
-        SimTK_INDEXCHECK(index, getNumConnectees(),
+        SimTK_INDEXCHECK(index, static_cast<int>(getNumConnectees()),
```

Comment:
These changes are because newer MSVCs find the `int` vs `unsigned` comparisons ambiguous.